lib/httpoison.ex

```diff
@@ -37,11 +37,11 @@ defmodule HTTPoison.Request do
   defstruct method: :get, url: nil, headers: [], body: "", params: %{}, options: []
 
   @type method :: :get | :post | :put | :patch | :delete | :options | :head
-  @type headers :: [{atom, binary}] | [{binary, binary}] | %{binary => binary}
-  @type url :: binary
-  @type body :: binary | charlist | iodata | {:form, [{atom, any}]} | {:file, binary}
-  @type params :: map | keyword | [{binary, binary}]
-  @type options :: keyword
+  @type headers :: [{atom, binary}] | [{binary, binary}] | %{binary => binary} | any
+  @type url :: binary | any
```

Comment:
now that I'm reading this we will accept `any` for  the result of `process_request_url` right? 🤔

Reply:
@edgurgel it's either this or telling people to cast to string first and close #327 (and possibly tag `wontfix`)

Reply:
Oh I agree with any for the input. I'm wondering about the return being `url` as `any` is also acceptable 🤔

Reply:
@edgurgel with this change `url` includes `| any`  ;)

Reply:
However, I'm fine either way, just let me know which you want and where.

Reply:
That's fine then 👍